web-server/v0.4/components/ListView/ResultListView.js

```diff
@@ -35,39 +34,73 @@ class ResultListView extends React.Component {
     }, 1000);
   }
 
-  onSelectChange = (selectedRowKeys) => {
-    this.setState({ selectedRowKeys });
+  openNotificationWithIcon = (type) => {
+    notification[type]({
+      message: 'Please select two results for comparison.',
+      placement: 'bottomRight'
+    });
   }
 
-  getResultMetadata(key) {
-    return axios.get('http://es-perf44.perf.lab.eng.bos.redhat.com:9280/dsa.pbench.*/_search?source={ "query": { "match": { "run.name": "' + key + '" } }, "fields": [ "run.name", "run.config", "run.start_run", "run.end_run", "run.script" ], "sort": "_index" }', { cancelToken: this.cancelToken.token }).then(res => {
-      this.setState({results: this.state.results.concat({result: res.data.hits.hits[0].fields['run.name'][0], config: res.data.hits.hits[0].fields['run.config'][0], startRun: res.data.hits.hits[0].fields['run.start_run'][0], endRun: res.data.hits.hits[0].fields['run.end_run'][0]})});
-    });
+  onCompareResults = () => {
+    const { selectedRowKeys, results } = this.state;
+    const { controller } = this.props;
+      var selectedResults = [];
+      for (var item in selectedRowKeys) {
+        var result = results[selectedRowKeys[item]];
+        result["controller"] = controller;
+        selectedResults.push(results[selectedRowKeys[item]])
+      }
+      this.compareResults(selectedResults)
   }
 
-  componentWillMount() {
-    this.cancelToken = CancelToken.source();
+  onSelectChange = (selectedRowKeys) => {
+    this.setState({ selectedRowKeys });
   }
 
   componentDidMount() {
     this.setState({loading: true});
-    axios.get('http://es-perf44.perf.lab.eng.bos.redhat.com:9280/dsa.pbench.*/_search?search_type=count&source={ "query": { "match": { "run.controller": "' + this.props.controller + '" } }, "aggs": { "run": { "terms": { "field": "run.name", "size": 0 } } } }').then(res => {
-      const response = res.data.aggregations.run.buckets;
-      var results = [];
-      for (var item in response) {
-        results.push(this.getResultMetadata(response[item].key))
-      }
-      axios.all(results).then(axios.spread(function (results) {
-      }));
-      this.setState({loading: false});
-    }).catch(error => {
-      console.log(error);
-      this.setState({loading: false});
-    });
-  }
+    const { controller } = this.props;
+    let data = localStorage.getItem(controller);
 
-  componentWillUnmount() {
-    this.cancelToken.cancel('Operation canceled by the user.');
+    if (!data) {
+      axios.post('http://es-perf44.perf.lab.eng.bos.redhat.com:9280/_search', 
```

Comment:
The `/_search` does not appear to specify any indices here, can we fix that to reference the pbench run indices?